id_minter/src/main/scala/uk/ac/wellcome/platform/idminter/steps/IdEmbedder.scala

```diff
@@ -0,0 +1,28 @@
+package uk.ac.wellcome.platform.idminter.steps
+
+import com.google.inject.Inject
+import com.twitter.inject.Logging
+import uk.ac.wellcome.metrics.MetricsSender
+import uk.ac.wellcome.models.Work
+
+import scala.concurrent.Future
+
+class IdEmbedder @Inject()(metricsSender: MetricsSender,
```

Comment:
Would this be better off as a trait?

Reply:
It could be a trait but I'm not sure if it would be better. I think it would be kind of equivalent so I don't really have a preference. I did it this way 'cause it was what came more naturally. But I think that this class might eventually swallow the class that's using it, so I think we can leave it as is for now